validator/client/runner.go

```diff
@@ -62,25 +67,52 @@ func run(ctx context.Context, v Validator) {
 			log.Fatalf("Slasher is not ready: %v", err)
 		}
 	}
-	if err := v.WaitForChainStart(ctx); err != nil {
-		log.Fatalf("Could not determine if beacon chain started: %v", err)
-	}
-	if err := v.WaitForSync(ctx); err != nil {
-		log.Fatalf("Could not determine if beacon node synced: %v", err)
-	}
-	if err := v.WaitForActivation(ctx); err != nil {
-		log.Fatalf("Could not wait for validator activation: %v", err)
-	}
-	go v.ReceiveBlocks(ctx)
-
-	headSlot, err := v.CanonicalHeadSlot(ctx)
-	if err != nil {
-		log.Fatalf("Could not get current canonical head slot: %v", err)
+	ticker := time.NewTicker(backOffPeriod)
+	var headSlot uint64
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			err := v.WaitForChainStart(ctx)
+			if err != nil && strings.Contains(err.Error(), connectionIssueMsg) {
+				log.Warnf("Could not determine if beacon chain started: %v", err)
+				continue
+			}
+			if err != nil {
+				log.Fatalf("Could not determine if beacon chain started: %v", err)
+			}
+			err = v.WaitForSync(ctx)
+			if err != nil && strings.Contains(err.Error(), connectionIssueMsg) {
+				log.Warnf("Could not determine if beacon chain started: %v", err)
+				continue
+			}
+			if err != nil {
+				log.Fatalf("Could not determine if beacon node synced: %v", err)
+			}
+			err = v.WaitForActivation(ctx)
+			if err != nil && strings.Contains(err.Error(), connectionIssueMsg) {
+				log.Warnf("Could not wait for validator activation: %v", err)
+				continue
+			}
+			if err != nil {
+				log.Fatalf("Could not wait for validator activation: %v", err)
+			}
+			headSlot, err = v.CanonicalHeadSlot(ctx)
+			if err != nil && strings.Contains(err.Error(), connectionIssueMsg) {
+				log.Warnf("Could not get current canonical head slot: %v", err)
+				continue
+			}
+			if err != nil {
+				log.Fatalf("Could not get current canonical head slot: %v", err)
+			}
+		}
+		break
 	}
+	connectionErrorChannel := make(chan error)
+	go v.ReceiveBlocks(ctx, connectionErrorChannel)
```

Comment:
there was no error handling for ReceiveBlocks while it was running in a separate goroutine